cmd/private-prow-configs-mirror/main.go

```diff
@@ -0,0 +1,437 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowconfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/plugins"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/promotion"
+)
+
+const (
+	openshiftPrivOrg = "openshift-priv"
+)
+
+type options struct {
+	ciOpConfigDir string
+	prowConfig    string
+	prowPlugins   string
+}
+
+type orgReposWithOfficialImages map[string]sets.String
+
+func (o orgReposWithOfficialImages) isOfficialRepo(org, repo string) bool {
+	if _, ok := o[org]; ok {
+		if o[org].Has(repo) {
+			return true
+		}
+	}
+	return false
+}
+
+func (o *orgReposWithOfficialImages) isOfficialRepoFull(orgRepo string) bool {
+	if orgRepoList := strings.Split(orgRepo, "/"); len(orgRepoList) > 1 {
+		return o.isOfficialRepo(orgRepoList[0], orgRepoList[1])
+	}
+	return false
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.ciOpConfigDir, "ci-op-config-path", "", "Path to directory containing ci-operator configurations")
+	fs.StringVar(&o.prowConfig, "prow-config", "", "Prow's configuration file")
+	fs.StringVar(&o.prowPlugins, "prow-plugins", "", "Prow's plugins configuration file")
+
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func (o *options) validate() error {
+	if len(o.ciOpConfigDir) == 0 {
+		return errors.New("--ci-op-config-path is not defined")
+	}
+
+	if len(o.prowConfig) == 0 {
+		return errors.New("--prow-config is not defined")
+	}
+
+	if len(o.prowPlugins) == 0 {
+		return errors.New("--prow-plugins is not defined")
+	}
+	return nil
+}
+
+func loadProwPlugins(pluginsPath string) (*plugins.Configuration, error) {
+	agent := plugins.ConfigAgent{}
+	if err := agent.Load(pluginsPath, false); err != nil {
+		return nil, err
+	}
+	return agent.Config(), nil
+}
+
+func updateProwConfig(configFile string, config prowconfig.ProwConfig) error {
+	data, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("could not marshal Prow configuration: %v", err)
+	}
+
+	return ioutil.WriteFile(configFile, data, 0644)
+}
+
+func updateProwPlugins(pluginsFile string, config *plugins.Configuration) error {
+	data, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("could not marshal Prow configuration: %v", err)
+	}
+
+	return ioutil.WriteFile(pluginsFile, data, 0644)
+}
+
+func privateOrgRepo(repo string) string {
+	return fmt.Sprintf("%s/%s", openshiftPrivOrg, repo)
+}
+
+func getOrgReposWithOfficialImages() (map[string]sets.String, error) {
```

Comment:
We defined a type `orgReposWithOfficialImages`, we should use it in the signature